pallets/asset/src/lib.rs

```diff
@@ -1363,6 +1367,26 @@ decl_module! {
             Self::deposit_event(RawEvent::ExtensionUnArchived(my_did, ticker, extension_id));
             Ok(())
         }
+
+        /// Sets the treasury DID to a given value. The caller must be the ticker owner.
+        ///
+        /// # Arguments
+        /// * `origin` - Signatory who owns the ticker/asset.
+        /// * `ticker` - Ticker symbol of the asset.
+        /// * `treasury_did` - The treasury DID wrapped in a value of type [`Option`].
+        #[weight = SimpleDispatchInfo::FixedNormal(250_000)]
+        pub fn set_treasury_did(
+            origin,
+            ticker: Ticker,
+            treasury_did: Option<IdentityId>,
+        ) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+            let did = Context::current_identity_or::<Identity<T>>(&sender)?;
+            ensure!(Self::is_owner(&ticker, did), Error::<T>::Unauthorized);
+            <Tokens<T>>::mutate(&ticker, |token| token.treasury_did = treasury_did);
+            Self::deposit_event(RawEvent::TreasuryDidSet(did, ticker, treasury_did));
```

Comment:
Do we want to allow the issuer to set the treasury account to `None or any other DID` when it posses the asset? Asset possed by that DID becomes immovable because the above behavior but this will not be the end of the world as issuer can easily add the claim to that identity to make those asset movable.

It is better to document this behavior in code comments and does the product team know about this kind of behaviour ?

Reply:
cc @adamdossa for visibility

Reply:
Here is the requirement: "the asset issuer can always update the treasuryDID, including setting it to None."

Let me know if that needs to be changed.

Reply:
I've added a doc comment with a copy of the requirement.

Reply:
Yep - this logic looks good to me. If an issuer modifies their treasuryDID to None, then as @satyamakgec says it will be locked until either they re-add it as the TreasuryDID, or they add a claim / rule etc. to allow this DID to move the asset.

Reply:
I added more explanations to the doc comment.